src/com/google/cose/utils/CoseUtils.java

```diff
@@ -45,4 +66,93 @@ public static DataItem encodeStructure(String context, Map protectedBodyHeaders,
+   * @return value in the map corresponding to key
+   */
+  public static DataItem getValueFromMap(Map cborMap, int index) {
+    final Number key;
+    if (index >= 0) {
+      key = new UnsignedInteger(index);
+    } else {
+      key = new NegativeInteger(index);
+    }
+    return cborMap.get(key);
+  }
+
+  /**
+   * Generates ECC Private Key from d coordinate value.
+   *
+   * Only supports P256 curve currently.
+   * @param curve supported curve
+   * @param d raw bytes for the private key coordinate.
+   * @return PrivateKey JCA implementation
+   * @throws CoseException if unsupported key curve is used.
+   */
+  public static PrivateKey generateEccPrivateKey(int curve, BigInteger d) throws CoseException {
+    try {
+      if (d == null) {
+        return null;
+      }
+      final AlgorithmParameters parameters = AlgorithmParameters.getInstance(EC_PARAMETER_SPEC);
+      parameters.init(getECCParameterSpecFromCurve(curve));
+      final ECParameterSpec ecParameters = parameters.getParameterSpec(ECParameterSpec.class);
+      final ECPrivateKeySpec privateKeySpec = new ECPrivateKeySpec(d, ecParameters);
+      return KeyFactory.getInstance(EC_PARAMETER_SPEC).generatePrivate(privateKeySpec);
+    } catch (final NoSuchAlgorithmException | InvalidParameterSpecException
+        | InvalidKeySpecException ex) {
+      throw new IllegalStateException("Unexpected error", ex);
+    }
+  }
+
+  /**
+   * Generates ECC Public Key from x and y coordinate values.
+   *
+   * Only supports P256 curve currently.
+   * @param curve supported curve
+   * @param x raw bytes for x coordinate.
+   * @param y raw bytes for y coordinate.
+   * @return PublicKey JCA implementation
+   * @throws CoseException if unsupported key curve is used.
+   */
+  public static PublicKey generateEccPublicKey(int curve, BigInteger x, BigInteger y)
+      throws CoseException {
+    try {
+      if (x == null || y == null) {
+        // Should not reach here since we should be able to catch it during decoding key.
+        throw new CoseException("Cannot decode key. Missing coordinate information.");
+      }
+      final AlgorithmParameters params = AlgorithmParameters.getInstance(EC_PARAMETER_SPEC);
+      params.init(getECCParameterSpecFromCurve(curve));
+      final ECParameterSpec ecParameters = params.getParameterSpec(ECParameterSpec.class);
+
+      final ECPoint ecPoint = new ECPoint(x, y);
+      final ECPublicKeySpec keySpec = new ECPublicKeySpec(ecPoint, ecParameters);
+      return KeyFactory.getInstance(EC_PARAMETER_SPEC).generatePublic(keySpec);
+    } catch (final InvalidParameterSpecException | InvalidKeySpecException
+        | NoSuchAlgorithmException ex) {
+      throw new IllegalStateException("Unexpected error", ex);
+    }
+  }
+
+  private static ECGenParameterSpec getECCParameterSpecFromCurve(int curve) throws CoseException {
+    if (curve == Headers.CURVE_EC2_P256) {
+      return new ECGenParameterSpec("prime256v1");
+    } else {
```

Comment:
I might recommend going ahead and adding the other NIST curves here, P384 and P521, just since it should be pretty easy to go ahead and plug them in here.